contracts/modules/TransferManager/MaximumVolumeTransferManager.sol

```diff
@@ -0,0 +1,290 @@
+pragma solidity ^0.4.24;
+
+import "./ITransferManager.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Transfer Manager module for capping the amount of token transfers within a period
+ */
+contract MaximumVolumeTransferManager is ITransferManager {
+	
+	using SafeMath for uint256;
+
+    bytes32 public constant ADMIN = "ADMIN";
+
+    enum RollingIntervals { Daily, Weekly, Quarterly, Biannually, Annually }
+
+    struct MaximumVolumeRestriction{
+        uint256 maximumVolumeAllowed;
+        uint256 startTime;
+        uint256 endTime;
+        RollingIntervals rollingPeriodInterval;
+    }
+
+    MaximumVolumeRestriction[] public maximumVolumeRestrictions;
+
+    // mapping from address to RollingIntervals to periodIds to tokens transfered 
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256))) volumeTally;
+
+    // Emit whenever a new maximum volume restriction gets added
+    event MaximumVolumeRestrictionAdded(
+        address indexed from,
+        uint256 timestamp,
+        uint256 _maximumVolume, 
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriodInterval
```

Comment:
emit the `index` of the new volume restriction

Reply:
done. adding an `_index` param to be the first